apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/eob/EobHandler.java

```diff
@@ -61,8 +70,41 @@ public Bundle searchByBene(
     var eobs =
         claimRepository.findByBeneXrefSk(
             beneXrefSk.get(), serviceDate, lastUpdated, count, startIndex, sourceIds);
+
+    return getFhirEobs(eobs, true);
+  }
+
+  Bundle getFhirEobs(List<Claim> eobs, boolean filterSamhsa) {
+
+    var filteredEobs = eobs;
+    var dict = SECURITY_LABELS_DICT;
+
+    if (filterSamhsa) {
+
+      var sys = SecurityLabels.getSecurityLabelKeys(dict);
+
+      filteredEobs =
+          eobs.stream()
+              // keep claims that do NOT contain any SAMHSA-coded procedures
+              .filter(
+                  claim -> {
+                    return !claimHasSamhsa(claim, dict, sys);
+                  })
+              .toList();
+    }
+
+    if (filteredEobs.isEmpty()) {
+      Bundle emptyBundle = new Bundle();
+      emptyBundle.setType(Bundle.BundleType.COLLECTION);
+      emptyBundle.setTypeElement(null);
+
+      return emptyBundle;
+    }
+
+    var fhirEobs = filteredEobs.stream().map(Claim::toFhir).toList();
+
     return FhirUtil.bundleOrDefault(
-        eobs.stream().map(Claim::toFhir), claimRepository::claimLastUpdated);
+        fhirEobs.stream().map(eob -> (Resource) eob), claimRepository::claimLastUpdated);
   }
```

Comment:
This `getFhirEobs` method mixes together logic. IMO it would be easier to read if some of it was split out into separate methods. Perhaps one for `filterSamhsaClaims()` then one for `createEmptyBundle()`?